test/common/upstream/load_balancer_impl_test.cc

```diff
@@ -2787,20 +2787,20 @@ TEST_P(LeastRequestLoadBalancerTest, SingleHost) {
 
   // Host weight is 1.
   {
-    EXPECT_CALL(random_, random()).WillOnce(Return(0)).WillOnce(Return(2)).WillOnce(Return(3));
+    EXPECT_CALL(random_, random()).WillOnce(Return(0)).WillOnce(Return(2));
     EXPECT_EQ(hostSet().healthy_hosts_[0], lb_.chooseHost(nullptr));
   }
 
   // Host weight is 100.
   {
-    EXPECT_CALL(random_, random()).WillOnce(Return(0)).WillOnce(Return(2)).WillOnce(Return(3));
+    EXPECT_CALL(random_, random()).WillOnce(Return(0)).WillOnce(Return(2));
     EXPECT_EQ(hostSet().healthy_hosts_[0], lb_.chooseHost(nullptr));
   }
 
```

Comment:
Based on my comments about only changing behavior when FULL_SCAN is set, I don't think you need to modify any existing unit tests. All that should be needed is a new one that sets the new selection method and verifies sane behavior.

You might even be able to get away with unit testing _just_ the selection method if it's in its own function.